modules/module.js

```diff
@@ -12,3 +12,14 @@ exports.fetchAllEndponts = () => {
     return JSON.parse(String(result));
   });
 };
+exports.fetchArticlById = (article_id) => {
+  return db
+    .query('SELECT * FROM articles WHERE article_id = $1;', [article_id])
+    .then((result) => {
+      if (!result.rows[0]) {
+        return Promise.reject({ msg: 'article does not exist' });
+      }
+
+      return result.rows[0];
+    });
+};
```

Comment:
Good work here too:

- Well named function name `fetchArticleById`
- SQL statement is spot on and great to see you used a parameterised query to avoid SQL injection.
- The handling of an empty result/array and rejecting it is spot on, well done! You might want to consider extending this Promise.reject to also send back a status code with the msg, that was you can extend the functionality of your error handling middleware in `app.js` - see my comment above with the nc notes link.